File: opn_test_template/lib/app/authentification/auth/model/wp_exception.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'wp_exception.g.dart';

@JsonSerializable()
class WpException implements Exception {
  WpException({
    required this.status,
    required this.reason,
    required this.errorMsg,
    required this.errorData,
  });

  factory WpException.fromJson(Map<String, dynamic> json) =>
      _$WpExceptionFromJson(json);

  Map<String, dynamic> toJson() => _$WpExceptionToJson(this);

  final int status;
  final String errorMsg;
  final String? reason;
  final String? errorData;

  bool get isInvalidCredentials =>
      status == 403 ||
      reason?.toLowerCase().contains('invalid_credentials') == true;

  bool get isServerError => status >= 500;
}

File: supabase_guardia_civil/functions/login-register/wp_client/model/wp_exception.ts
// wp_exception.ts
export class WpException extends Error {
  status: number;
  reason?: string;
  errorMsg: string;
  errorData?: string;

  constructor(
    status: number = 0,
    errorMsg: string = '',
    reason?: string,
    errorData?: string
  ) {
    super(errorMsg);
    this.name = 'WpException';
    this.status = status;
    this.reason = reason;
    this.errorMsg = errorMsg;
    this.errorData = errorData;
  }

  static fromJson(json: any): WpException {
    return new WpException(
      json.status || 0,
      json.errorMsg || '',
      json.reason,
      json.errorData
    );
  }

  toJson(): any {
    return {
      status: this.status,
      reason: this.reason,
      errorMsg: this.errorMsg,
      errorData: this.errorData,
    };
  }
}